### PR TITLE
Framework: Replace React.PropTypes with PropTypes

### PR DESCRIPTION
**Pull request by jsnmoon**
Part of #13321; an updated take on #18069.

This replaces all instances of React.PropTypes with PropTypes from `'prop-types'` package. This is known to be a safe transformation. In other words, no product testing should necessary - if it builds, it'd work.

I reviewed each file individually before opening this PR, but if you see something amiss (like an extra newline or mis-arranged imports), please let me know! 

I would like to ship this quickly in order to avoid merge conflicts.

### CODE REVIEW

**Comment by samouri** [comment]
How did you remove all the automatically added newlines?

**Comment by samouri** [comment]
About 10 files currently have `need space inside parens` issue.  Might be worth fixing them all to make CircleCI happy

File: client/me/security-account-recovery/edit-email.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /**
  * External dependencies
  */
+var PropTypes = require('prop-types');
```
Comment (spen) [inline client/me/security-account-recovery/edit-email.jsx]: So far so good!
I noticed that some imports use `var`, a little weird but that's hardly worth blocking this :)

Comment (jsnmoon) [reply client/me/security-account-recovery/edit-email.jsx]: I figured we could fix this with the imports codemod after the fact :)

**Comment by jsnmoon** [comment]
> How did you remove all the automatically added newlines?

I used a personal regex script to kill the extra lines. Not exactly kosher to commit to the repo proper, I found out ;)

**Comment by ockham** [comment]
> I reviewed each file individually before opening this PR

image

😱

**Comment by ockham** [comment]
Awesome! Great work @jsnmoon and @samouri 😄

**Pull request merged**